slack-base/src/main/java/com/hubspot/slack/client/models/dialog/form/elements/AbstractSlackFormSelectElement.java

```diff
@@ -40,56 +41,61 @@ public SlackDataSource getDataSource() {
   public abstract Optional<Integer> getMinQueryLength();
 
   @Check
-  public void validate() {
-    super.validateBaseElementProperties();
+  public AbstractSlackFormSelectElement validate() {
+    AbstractSlackFormSelectElement normalized = SlackDialogElementNormalizer.normalize(this);
+    super.validateBaseElementProperties(normalized);
     List<String> errors = new ArrayList<>();
 
-    int numOptions = getOptions().size();
-    int numOptionGroups = getOptionGroups().size();
+    List<SlackFormOption> normalizedOptions = normalized.getOptions();
+    int numOptions = normalizedOptions.size();
+    List<SlackFormOptionGroup> normalizedOptionGroups = normalized.getOptionGroups();
+    int numOptionGroups = normalizedOptionGroups.size();
 
-    if (numOptions > 100) {
+    if (numOptions > SlackDialogFormElementLengthLimits.MAX_OPTIONS_NUMBER.getLimit()) {
       errors.add("Cannot have more than 100 options");
```

Comment:
Can we use limit values from enum here and in all other logging statements? I think we can update it in the enum and forget to update it here.

Reply:
Good call 👍